apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsDropdownControl.tsx

```diff
@@ -28,7 +28,17 @@ export function JsonFormsDropdownControl({
 }: ControlProps & OwnPropsOfEnum) {
   const [dropdownValue, setDropdownValue] = useState(data || "")
 
-  if (!options || (options.length === 1 && !!schema.default)) {
+  // Use the default value if it exists
+  useEffect(() => {
+    if (data !== undefined || data !== "") {
+      return
+    }
+    const value = schema.default || schema.const
+    setDropdownValue(value)
+    handleChange(path, value)
+  }, [path, schema.default, schema.const, handleChange, data])
```

Comment:
[don't sync state, derive it!](https://www.notion.so/opengov/Don-t-sync-state-derive-it-173d074b374644e6bab52e578ad626a3)

feels weird to need this, does JSONForm not update something like `data` when  `handleChange` is triggered? if so, can't you derive the current selected value from it? unsure how it actually works.

```ts
const dropdownValue = data ?? "" // this should always be synchronised with the prop

// unsure if you ever need to call handleChange in an effect?

// then inside singleselect
<SingleSelect
  value={dropdownValue}
  onChange={value => handleChange(path, value)} // < this should update data prop, no?
/>
```

Reply:
Ah yup I think I forgot to remove this earlier, done in 993b9c92c9f4d1c3a6a5405ef52321df66759a0e.